metaflow/cli_components/run_cmds.py

```diff
@@ -380,3 +395,126 @@ def run(
+    "corresponding to this task ID. If not provided, an arbitrary task ID from the latest "
+    "run will be used.",
+)
+@click.option(
+    "--skip-decorators/--no-skip-decorators",
+    is_flag=True,
+    default=False,
+    show_default=True,
+    help="Skip decorators attached to the step.",
+)
+@click.option(
+    "--artifacts-module",
+    default=None,
+    show_default=True,
+    help="Path to a module that contains artifacts to be used in the spun step. The artifacts should "
+    "be defined as a dictionary called ARTIFACTS with keys as the artifact names and values as the "
+    "artifact values. The artifact values will overwrite the default values of the artifacts used in "
+    "the spun step.",
+)
+@click.option(
+    "--max-log-size",
+    default=10,
+    show_default=True,
+    help="Maximum size of stdout and stderr captured in "
+    "megabytes. If a step outputs more than this to "
+    "stdout/stderr, its output will be truncated.",
+)
+@common_runner_options
+@click.pass_obj
+def spin(
+    obj,
+    step_name=None,
+    spin_pathspec=None,
+    artifacts_module=None,
+    skip_decorators=False,
+    max_log_size=None,
+    run_id_file=None,
+    runner_attribute_file=None,
+    **kwargs
+):
+    before_run(obj, [], [])
+    if step_name and spin_pathspec:
+        raise CommandException(
+            "Cannot specify both step-name and spin-pathspec. Please specify only one."
+        )
+    if not (step_name or spin_pathspec):
+        raise CommandException(
+            "Please specify either step-name or spin-pathspec to spin a task."
+        )
+    if step_name is not None:
+        spin_pathspec = get_latest_task_pathspec(obj.flow.name, step_name)
+    else:
+        step_name = spin_pathspec.split("/")[2]
+
+    obj.echo(
+        f"Spinning up step *{step_name}* locally using previous task pathspec *{spin_pathspec}*"
+    )
+    obj.flow._set_constants(obj.graph, kwargs, obj.config_options)
+    step_func = getattr(obj.flow, step_name)
+
+    spin_runtime = SpinRuntime(
+        obj.flow,
+        obj.graph,
+        obj.flow_datastore,
+        obj.metadata,
+        obj.environment,
+        obj.package,
+        obj.logger,
+        obj.entrypoint,
+        obj.event_logger,
+        obj.monitor,
+        obj.spin_metadata,
+        obj.spin_flow_datastore,
+        step_func,
+        spin_pathspec,
+        skip_decorators,
+        artifacts_module,
+        max_log_size * 1024 * 1024,
+    )
+    _system_logger.log_event(
+        level="info",
+        module="metaflow.task",
+        name="spin",
+        payload={
+            "msg": str(
+                {
+                    "step_name": step_name,
+                    "task_pathspec": spin_pathspec,
+                }
+            )
+        },
+    )
+
+    write_latest_run_id(obj, spin_runtime.run_id)
+    write_file(run_id_file, spin_runtime.run_id)
+    spin_runtime.execute()
+
+    if runner_attribute_file:
```

Comment:
Shouldn't this come before the `execute` bit (else it makes it hard for async stuff to work).